src/flows/shared/steps/OrganizeCollection/Sidebar/SidebarChainSelector.tsx

```diff
@@ -0,0 +1,226 @@
+   */
+  const [refresh] = usePromisifiedMutation<SidebarChainSelectorMutation>(graphql`
+    mutation SidebarChainSelectorMutation($chain: Chain!) {
+      syncTokens(chains: [$chain]) {
+        __typename
+        ... on SyncTokensPayload {
+          viewer {
+            ...CollectionEditorViewerFragment
+          }
+        }
+      }
+    }
+  `);
+
+  const reportError = useReportError();
+  const { pushToast } = useToastActions();
+  const [showTooltip, setShowTooltip] = useState(false);
+
+  const is3ac = isViewerId3ac(query.viewer?.user?.dbid);
+  const selectedChain = chains.find((chain) => chain.name === selected);
+
+  const handleRefresh = useCallback(async () => {
+    if (!selectedChain) {
+      return;
+    }
+
+    setIsRefreshingNfts(true);
+
+    pushToast({
+      message: 'We’re retrieving your new pieces. This may take up to a few minutes.',
+      autoClose: true,
+    });
+
+    try {
+      const response = await refresh({
+        variables: {
+          chain: selectedChain.name,
+        },
+      });
+
+      if (response.syncTokens?.__typename !== 'SyncTokensPayload') {
+        pushToast({
+          autoClose: false,
+          message:
+            'There was an error while trying to sync your tokens. We have been notified and are looking into it.',
+        });
+
+        reportError('Error while syncing tokens for chain. Typename was not `SyncTokensPayload`', {
+          level: Severity.Error,
+          tags: {
+            chain: selectedChain.name,
+            responseTypename: response.syncTokens?.__typename,
+          },
+        });
+      }
+    } catch (e) {
+      if (e instanceof Error) {
+        reportError(e);
+      } else {
+        reportError('Could not run SidebarChainSelectorMutation for an unknown reason');
+      }
+    } finally {
+      setIsRefreshingNfts(false);
+    }
+  }, [pushToast, refresh, reportError, selectedChain, setIsRefreshingNfts]);
+
+  if (!selectedChain) {
+    throw new Error(`Could not find a chain for selected value '${selected}'`);
+  }
+
+  return (
+    <Container>
+      <Chains>
+        {chains.map((chain) => {
+          const isSelected = chain.name === selected;
+
+          return (
+            <ChainButton
+              selected={isSelected}
+              role="button"
+              key={chain.name}
+              onClick={() => onChange(chain.name)}
+            >
+              <ChainLogo src={chain.icon} />
+              <TitleXSBold>{chain.shortName}</TitleXSBold>
+            </ChainButton>
+          );
+        })}
+      </Chains>
+      {!is3ac && (
+        <IconButton
+          refreshing={isRefreshingNfts}
+          data-testid="RefreshButton"
+          onMouseEnter={() => setShowTooltip(true)}
+          onMouseLeave={() => setShowTooltip(false)}
+          onClick={handleRefresh}
+        >
+          <StyledIconContainer icon={<RefreshIcon />} />
+          <RefreshTooltip
```

Comment:
nit: if I hit `tab` in the search box, it shifts focus to the refresh button which highlights the invisible text and looks a bit odd. not a blocker


![Screen Shot 2022-09-12 at 20 31 58](https://user-images.githubusercontent.com/80802871/189643153-9708abaa-8a3d-4b83-ad36-630c805dad45.png)